### PR TITLE
Prepare for publishing on https://hex.pm

### PR DESCRIPTION
**Pull request by rpelyush**
Add documentation in scope of this PR.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -33,4 +36,12 @@ defmodule Gardien.Mixfile do
       {:plug, "~> 1.1.2"}
     ]
   end
+
+  defp package do
+    %{
+      maintainers: ["Roman Peliushkevych"],
```
Comment (rpelyush) [inline mix.exs]: @banzay don't forget to add your name here ;)

File: lib/gardien.ex [hunk 2]
```diff
@@ -1,8 +1,55 @@
 defmodule Gardien do
+  @moduledoc """
```
Comment (rpelyush) [inline lib/gardien.ex]: @banzay I would appreciate your feedback on module documentation.

File: lib/gardien.ex [hunk 3]
```diff
@@ -1,8 +1,79 @@
...
```
Comment (rpelyush) [inline lib/gardien.ex]: @banzay could you please review module documentation?

Comment (rpelyush) [inline lib/gardien.ex]: I see that you're writing alternative docs version (https://github.com/rpelyush/gardien/commits/init-hex-kissed) :)
Could you please write your suggestions here so we could discuss them in scope of this PR?
You can, for example, leave you comments on lines you'd like to change.

Comment (rpelyush) [inline lib/gardien.ex]: And then we'll work on README

File: lib/gardien.ex [hunk 4]
```diff
@@ -1,8 +1,79 @@
...
+
+  Before you plug in Gardien, please consider the following:
+  - Gardien doesn't handle user authentication. It expects user to be already authenticated.
+    Gardien will look for user in `conn.assigns` by specified key. This can be configured via env. config or passed as an option to plug.
+    If `current_user` is not configured and `current_user` option is missing, default key `:current_user` is assumed.
+    In case user is missing in `conn.assigns`, action is considered to be unauthorized!
+  - Gardien doesn't know how to load resources from db and expects you to handle this.
+    It will look for resource, whose authorization you want to check, in `conn.assigns` by specified key.
+    In case you don't have a particular resource to authorize you can pass a simple struct, e.g `%Post{}`, directly through options.
+  - Gardien relies on policies, modules containing authorization logic (see policy example below).
```
Comment (banzay) [inline lib/gardien.ex]: policies - modules

File: lib/gardien.ex [hunk 5]
```diff
@@ -1,8 +1,79 @@
...
+          resource.user_id == user.id
+        end
+      end
+
+  What is policy:
+  - Module that exports query functions, in this case `update`.
+    Usually this should map to the name of a particular controller action.
+  - Each query function takes two arguments: `user` performing an action and some kind of `resource`.
+  - Each query function should return `true` or `false`/`nil`.
+
```
Comment (banzay) [inline lib/gardien.ex]: We suggest you putting policies

File: lib/gardien.ex [hunk 6]
```diff
@@ -1,8 +1,79 @@
...
+    In case `unauth_handler` is missing, default handler is assumed.
+
+  ## Policy example:
+
+      module PostPolicy do
+        def update(user, resource) do
+          resource.user_id == user.id
+        end
+      end
+
```
Comment (banzay) [inline lib/gardien.ex]: What is a policy

File: lib/gardien.ex [hunk 7]
```diff
@@ -1,8 +1,79 @@
...
+  What is policy:
+  - Module that exports query functions, in this case `update`.
```
Comment (banzay) [inline lib/gardien.ex]: Usually these functions should map to the names of a particular controller actions.

File: lib/gardien.ex [hunk 8]
```diff
@@ -1,8 +1,79 @@
...
+    Usually this should map to the name of a particular controller action.
+  - Each query function takes two arguments: `user` performing an action and some kind of `resource`.
+  - Each query function should return `true` or `false`/`nil`.
+
+  We suggest to put policies into `/web/policies` folder.
+
+  Gardien is implemented as a module plug (exports `call/2` and `init/1` functions).
```
Comment (banzay) [inline lib/gardien.ex]: Gardien uses module plug's ability to validate arguments during compilation.

File: lib/gardien.ex [hunk 9]
```diff
@@ -1,8 +1,79 @@
...
+
+  Supported options:
+  - `resource` - atom, function or struct (required).
+  - `policy` (as described above) - module (required).
+  - `current_user` - atom (optional).
+    Can be configured via env config. By default Gardien assumes `:current_user`.
+  - `unauth_handler` - function or `{Module, atom}` tuple (optional).
+    Can be configured via env config. `unauth_handler` should be a function that takes `conn` as an argument and returns `conn`.
```
Comment (banzay) [inline lib/gardien.ex]: default is assumed (403 Forbidden) -> default handler will return 403

File: lib/gardien.ex [hunk 10]
```diff
@@ -1,8 +1,79 @@
...
+  What is policy:
+  - Module that exports query functions, in this case `update`.
+    Usually this should map to the name of a particular controller action.
+  - Each query function takes two arguments: `user` performing an action and some kind of `resource`.
+  - Each query function should return `true` or `false`/`nil`.
+
+  We suggest to put policies into `/web/policies` folder.
+
+  Gardien is implemented as a module plug (exports `call/2` and `init/1` functions).
+
```
Comment (banzay) [inline lib/gardien.ex]: I'd put a title `## Plug usage ` here

File: lib/gardien.ex [hunk 11]
```diff
@@ -1,8 +1,79 @@
...
+      end
+
+  What is policy:
+  - Module that exports query functions, in this case `update`.
+    Usually this should map to the name of a particular controller action.
+  - Each query function takes two arguments: `user` performing an action and some kind of `resource`.
+  - Each query function should return `true` or `false`/`nil`.
+
+  We suggest to put policies into `/web/policies` folder.
+
```
Comment (banzay) [inline lib/gardien.ex]: I'd put a title `## Plug usage ` here

File: lib/gardien.ex [hunk 12]
```diff
@@ -1,8 +1,79 @@
 defmodule Gardien do
+  @moduledoc """
+  Module that provides policy based authorization for Phoenix applications.
+
```
Comment (banzay) [inline lib/gardien.ex]: Put `## Optional configuration` after Before you plug....

**Comment by banzay** [comment]
Good job. I don't have much to add

File: lib/gardien.ex [hunk 13]
```diff
@@ -1,8 +1,79 @@
...
+  - Module that exports query functions, in this case `update`.
+    Usually this should map to the name of a particular controller action.
+  - Each query function takes two arguments: `user` performing an action and some kind of `resource`.
+  - Each query function should return `true` or `false`/`nil`.
+
+  We suggest to put policies into `/web/policies` folder.
+
+  Gardien is implemented as a module plug (exports `call/2` and `init/1` functions).
+
+  Supported options:
```
Comment (banzay) [inline lib/gardien.ex]: atom, function or struct (required).
If resource is not needed to authorize, you can pass empty struct.

File: lib/gardien.ex [hunk 14]
```diff
@@ -1,8 +1,79 @@
...
+        unauth_handler: fn(conn) -> conn end
+
+  Before you plug in Gardien, please consider the following:
+  - Gardien doesn't handle user authentication. It expects user to be already authenticated.
+    Gardien will look for user in `conn.assigns` by specified key. This can be configured via env. config or passed as an option to plug.
+    If `current_user` is not configured and `current_user` option is missing, default key `:current_user` is assumed.
+    In case user is missing in `conn.assigns`, action is considered to be unauthorized!
+  - Gardien doesn't know how to load resources from db and expects you to handle this.
+    It will look for resource, whose authorization you want to check, in `conn.assigns` by specified key.
+    In case you don't have a particular resource to authorize you can pass a simple struct, e.g `%Post{}`, directly through options.
```
Comment (rpelyush) [inline lib/gardien.ex]: https://github.com/rpelyush/gardien/issues/36

**Comment by rpelyush** [comment]
I suppose we can add README in another PR. Module documentation is enough for publishing on hex.

**Pull request merged**